server/app/services/scheduled_nps_report_service.py

```
@@ -0,0 +1,455 @@
+
+class ScheduledNPSReportService:
+    """Service for managing scheduled NPS reports."""
+
+    @staticmethod
+    def _calculate_next_scheduled_time(
+        frequency: ReportFrequency,
+        scheduled_hour: int = 9,
+        scheduled_minute: int = 0,
+        day_of_week: Optional[int] = None,
+        day_of_month: Optional[int] = None,
+        current_time: datetime = None,
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)